examples/plot_pp_study.py

```diff
@@ -0,0 +1,362 @@
+# (described here). Originally, this dataset includes the following variables:
+#
+# .. list-table::
+#    :header-rows: 1
+#    :widths: 5 20
+#    :stub-columns: 1
+#
+#    *  - Variable
+#       - Description
+#    *  - CRIM (BLUE)
+#       - per capita crime rate by town
+#    *  - ZN
+#       - proportion of residential land zoned for lots over 25,000 sq.ft.
+#    *  - INDUS
+#       - proportion of non-retail business acres per town
+#    *  - CHAS
+#       - Charles River dummy variable (= 1 if tract bounds river; 0 otherwise)
+#    *  - NOX
+#       - nitric oxides concentration (parts per 10 million)
+#    *  - RM
+#       - average number of rooms per dwelling
+#    *  - AGE
+#       - proportion of owner-occupied units built prior to 1940
+#    *  - DIS
+#       - weighted distances to five Boston employment centres
+#    *  - RAD
+#       - index of accessibility to radial highways
+#    *  - TAX
+#       - full-value property-tax rate per $10,000
+#    *  - PTRATIO
+#       - pupil-teacher ratio by town
+#    *  - B
+#       - 1000(Bk - 0.63)^2 where Bk is the proportion of blacks by town
+#    *  - LSTAT (BLUE)
+#       - % lower status of the population
+#    *  - MEDV (GREEN)
+#       - Median value of owner-occupied homes in $1000's
+# *Figure A: Dataset variables*
+#
+# Where MEDV (colored green in Figure A above) is the target variable
+# (that a model built on this dataset is intended to predict). We make
+# modifications below with the understanding that LSTAT is a parameter of
+# concern.
+#
+# **Import and install relevant packages**
+#
+import numpy as np
+import pandas as pd
+from fairlearn.metrics import MetricFrame
+from fairlearn.metrics import selection_rate
+from fairlearn.reductions import DemographicParity, ExponentiatedGradient
+from fairlearn.datasets import fetch_boston
+from sklearn.metrics import accuracy_score
+from sklearn.tree import DecisionTreeClassifier
+from statistics import mean, median
+from fairlearn.widget import FairlearnDashboard
+
+# %%
+#
+# **1. We devise a new target column based on CRIM (colored blue in Figure A above), indicating whether a certain per capita crime rate threshold is met.**
+#
+#    While we calculate and print Mean, Median, Max, and Min crime
+#    rates below, the analysis to follow is specifically for the mean crime
+#    rate as the chosen threshold. Similar results are observed for median
+#    crime rate as the chosen threshold as well (which one can observe by
+#    changing the index of threshold_crime_rates to 1).
+#
+bostonDS = fetch_boston(as_frame=True)
+crimerate_data = bostonDS.data['CRIM']
+
+mean_crimerate = mean(crimerate_data)
+median_crimerate = median(crimerate_data)
+threshold_crime_rates = [mean_crimerate, median_crimerate]
+min_crimerate = min(crimerate_data)
+max_crimerate = max(crimerate_data)
+"""
+[0] = Mean
+[1] = Median
+"""
+y_true = (crimerate_data > threshold_crime_rates[0]) * 1
```

Comment:
Interesting design choice. Why binary classification and not regression? I think this goes back to describing what would happen to the model outputs. Does 1 mean we'll have police presence there? It's not 100% clear to me. If it's a regression task then I can proportionately assign people to go to certain areas, but if I have way too many 1s then I don't know how I'd decide.
[I wouldn't want pred. policing to be used at all, so this is purely hypothetical, but I think it needs a bit more detail to make it clear what harms this introduces]

Reply:
Given that two of the LSTAT bins end up with 100% accuracy I very much agree. Results like that are suspicious :-)

Reply:
Might be worth looking at how this task is traditionally framed also. The CAPP-PGH primer might be helpful here. Depending on your goal, you may want to replicate what those approaches do, or take a different approach.